src/sage/schemes/generic/homset.py

```diff
@@ -658,6 +658,56 @@ def _element_constructor_(self, *v, **kwds):
             v = v[0]
         return self.extended_codomain()._point(self, v, **kwds)
 
+    def __iter__(self):
+        r"""
+        Return an iterator for the set of rational points on this scheme.
+
+        By default, this calls the :meth:`points` method, which is implemented
+        when the base ring is a field
+
+        - for affine homsets at :meth:`sage.schemes.affine.affine_homset.SchemeHomset_points_affine.points`;
+        - for projective homsets at :meth:`sage.schemes.projective.projective_homset.SchemeHomset_points_projective_field.points`;
+        - and toric homsets at :meth:`sage.schemes.toric.homset.SchemeHomset_points_toric_field._enumerator`.
+
+        OUTPUT: iterator over points
+
+        EXAMPLES::
```

Comment:
```suggestion
        TESTS::
```

Since `__iter__` isn't included in the docs, these aren't examples that the user will see so I think a `TESTS` block is more fitting.

I don't love the use of `_` in examples that users will see in case the user isn't familiar with that peculiarity of Python syntax, but since this won't be included in the documentation I think it's fine.